### PR TITLE
BoomBox

### PR DESCRIPTION
**Pull request by dragoslava**
Adds a boombox along with some new songs for it
dreamseeker_2018-10-25_21-19-41

Usage is as follows:

Carry with you, set down and use the set down verb. Alt Click it to pick a song, to move it again to take with you simply use the pick up boom box verb

### CODE REVIEW

**Comment by BotBOREALIS** [comment]
Icons with diff:
<details><summary>icons/mob/items/lefthand.dmi</summary>

Key | Old | New | Status
--- | --- | --- | ---
boombox||boombox|Created
boombox[1]||![boombox[1]](http://images.aurorastation.org/images/57cee8ac70aa5df818d7938602712e90.png)|Created
boombox[2]||![boombox[2]](http://images.aurorastation.org/images/f1f485d8187c99aeedd52beba2c1c7c5.png)|Created
boombox[3]||![boombox[3]](http://images.aurorastation.org/images/59f1902424179c290439853380926621.png)|Created
</details>
<details><summary>icons/mob/items/righthand.dmi</summary>

Key | Old | New | Status
--- | --- | --- | ---
boombox||boombox|Created
boombox[1]||![boombox[1]](http://images.aurorastation.org/images/6b01d246dcdd5827bd0c944378b14655.png)|Created
boombox[2]||![boombox[2]](http://images.aurorastation.org/images/8e953c0c79497b37a8522b5e6b26892c.png)|Created
boombox[3]||![boombox[3]](http://images.aurorastation.org/images/2db84af9f60a65d239085922a6966d2e.png)|Created
</details>
<details><summary>icons/obj/jukebox.dmi</summary>

Key | Old | New | Status
--- | --- | --- | ---
|||Created
boombox-nopower||boombox-nopower|Created
boombox-nopower[1]||![boombox-nopower[1]](http://images.aurorastation.org/images/96421467d17934b921a9430ff787b04e.png)|Created
boombox-nopower[2]||![boombox-nopower[2]](http://images.aurorastation.org/images/96421467d17934b921a9430ff787b04e.png)|Created
boombox-nopower[3]||![boombox-nopower[3]](http://images.aurorastation.org/images/96421467d17934b921a9430ff787b04e.png)|Created
boombox-nopower[4]||![boombox-nopower[4]](http://images.aurorastation.org/images/96421467d17934b921a9430ff787b04e.png)|Created
boombox-nopower[5]||![boombox-nopower[5]](http://images.aurorastation.org/images/96421467d17934b921a9430ff787b04e.png)|Created
boombox-nopower[6]||![boombox-nopower[6]](http://images.aurorastation.org/images/96421467d17934b921a9430ff787b04e.png)|Created
boombox-nopower[7]||![boombox-nopower[7]](http://images.aurorastation.org/images/44a0a4263dd0f78fd75218ace5f914ef.png)|Created
boombox-nopower[8]||![boombox-nopower[8]](http://images.aurorastation.org/images/96421467d17934b921a9430ff787b04e.png)|Created
boombox-nopower[9]||![boombox-nopower[9]](http://images.aurorastation.org/images/96421467d17934b921a9430ff787b04e.png)|Created
boombox-nopower[10]||![boombox-nopower[10]](http://images.aurorastation.org/images/96421467d17934b921a9430ff787b04e.png)|Created
boombox-running||boombox-running|Created
boombox-running[1]||![boombox-running[1]](http://images.aurorastation.org/images/0e66de83daa8fe18fb53e12830118263.png)|Created
boombox-running[2]||![boombox-running[2]](http://images.aurorastation.org/images/4ecafc48f7605fc7152ec3f39ebbfdbe.png)|Created
boombox-running[3]||![boombox-running[3]](http://images.aurorastation.org/images/1b5b139fe2cbeb3b2c50089eb79cfd3c.png)|Created
boombox-running[4]||![boombox-running[4]](http://images.aurorastation.org/images/71f5c562547c008489c810e076687bd3.png)|Created
boombox-running[5]||![boombox-running[5]](http://images.aurorastation.org/images/ce39fc054caa41a1582dad49d33c35f0.png)|Created
boombox-running[6]||![boombox-running[6]](http://images.aurorastation.org/images/834fe6dc413367610943af31ecd76bdd.png)|Created
boombox-running[7]||![boombox-running[7]](http://images.aurorastation.org/images/9af537e628e81ffcc99d74d0e05b39b5.png)|Created
boombox||boombox|Created
boombox[1]||![boombox[1]](http://images.aurorastation.org/images/96421467d17934b921a9430ff787b04e.png)|Created
boombox[2]||![boombox[2]](http://images.aurorastation.org/images/96421467d17934b921a9430ff787b04e.png)|Created
boombox[3]||![boombox[3]](http://images.aurorastation.org/images/96421467d17934b921a9430ff787b04e.png)|Created
boombox[4]||![boombox[4]](http://images.aurorastation.org/images/96421467d17934b921a9430ff787b04e.png)|Created
boombox[5]||![boombox[5]](http://images.aurorastation.org/images/96421467d17934b921a9430ff787b04e.png)|Created
boombox[6]||![boombox[6]](http://images.aurorastation.org/images/96421467d17934b921a9430ff787b04e.png)|Created
boombox[7]||![boombox[7]](http://images.aurorastation.org/images/44a0a4263dd0f78fd75218ace5f914ef.png)|Created
boombox[8]||![boombox[8]](http://images.aurorastation.org/images/96421467d17934b921a9430ff787b04e.png)|Created
boombox[9]||![boombox[9]](http://images.aurorastation.org/images/96421467d17934b921a9430ff787b04e.png)|Created
boombox[10]||![boombox[10]](http://images.aurorastation.org/images/96421467d17934b921a9430ff787b04e.png)|Created
</details>

**Comment by Lohikar** [comment]
You realize that some of these songs are copyrighted, yes?

File: aurorastation.dme [hunk 1]
```diff
@@ -1181,13 +1182,6 @@
 #include "code\modules\cargo\exports\sheets.dm"
 #include "code\modules\cargo\exports\tools.dm"
 #include "code\modules\cargo\exports\weapons.dm"
-#include "code\modules\cargo\random_stock\_defs.dm"
```
Comment (Lohikar) [inline aurorastation.dme]: You reverted these changes from another PR.

File: code/game/objects/items.dm [hunk 2]
```diff
@@ -454,6 +454,9 @@ var/list/global/slot_flags_enumeration = list(
 	if(I.anchored) //Object isn't anchored
 		usr << "<span class='warning'>You can't pick that up!</span>"
 		return
+	if(anchored)
+		usr << "<span class='warning'>You must secure \the item first.</span>"
```
Comment (Lohikar) [inline code/game/objects/items.dm]: So, if the mob is anchored?

File: code/game/objects/items/devices/beatbox.dm [hunk 3]
```diff
@@ -0,0 +1,202 @@
+// Updated Jukebox Music aswell, All songs are used are in Fair Use and credit is due to all authors
+
+
+datum/track
```
Comment (Lohikar) [inline code/game/objects/items/devices/beatbox.dm]: This already exists as part of the jukebox. Don't duplicate it.

File: code/game/objects/items/devices/beatbox.dm [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+
+datum/track/New(var/title_name, var/audio)
+	title = title_name
+	sound = audio
+
+/obj/item/device/boombox
+	name = "TuneMaster 5000"
+	desc = "A boombox, the annoyance of the jukebox in you're hands"
```
Comment (Lohikar) [inline code/game/objects/items/devices/beatbox.dm]: What is grammar.

File: code/game/objects/items/devices/beatbox.dm [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+/obj/item/device/boombox/attack_ai(mob/user as mob)
+	return src.attack_hand(user)
+
+/obj/item/device/boombox/attack_hand(mob/user as mob)
+	if(anchored)
+		usr << "<span class='warning'>You must un-secure \the [src] first.</span>"
+		return
+	..(user)
+
+/obj/item/device/boombox/AltClick(var/mob/user as mob)
```
Comment (Lohikar) [inline code/game/objects/items/devices/beatbox.dm]: butwhy

File: code/game/objects/items/devices/beatbox.dm [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+
+	if(!anchored)
+		usr << "<span class='warning'>You must secure \the [src] first.</span>"
+		return
+	if(anchored)
+		interact(user)
+
+/obj/item/device/boombox/proc/explode()
```
Comment (Lohikar) [inline code/game/objects/items/devices/beatbox.dm]: This doesn't seem to be used anywhere.

File: code/game/objects/items/devices/beatbox.dm [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+	visible_message("<span class='danger'>\the [src] blows apart!</span>")
+
+	explosion(src.loc, 0, 0, 1, rand(1,2), 1)
+
+	spark(src, 3, alldirs)
+
+	new /obj/effect/decal/cleanable/blood/oil(src.loc)
+	qdel(src)
+
+/obj/item/device/boombox/attackby(obj/item/W as obj, mob/user as mob)
```
Comment (Lohikar) [inline code/game/objects/items/devices/beatbox.dm]: Isn't this handled elsewhere in attack code already?

File: code/game/objects/items/devices/beatbox.dm [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+	if(istype(M) && M.a_intent == I_HURT)
+		walk_to(src,0)
+		visible_message("<span class='danger'>\the [src] blows apart!</span>")
+		spark(src, 3, alldirs)
+		new /obj/effect/decal/cleanable/blood/oil(src.loc)
+		playsound(loc, 'sound/effects/djscratch.ogg', 50, 1)
+		qdel(src)
+
+
+/obj/item/device/boombox/verb/SettleBox()
```
Comment (Lohikar) [inline code/game/objects/items/devices/beatbox.dm]: Missing sanity.

File: code/game/objects/items/devices/beatbox.dm [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+	set src in oview(1)
+	set category = "Object"
+	set name = "Set Boombox Up"
+	anchored = 1
+	usr.visible_message("<span class='notice'>\the [src] is setup to play music now.</span>")
+	
+
+
+/obj/item/device/boombox/verb/Unsettle()
```
Comment (Lohikar) [inline code/game/objects/items/devices/beatbox.dm]: Missing sanity.

**Pull request closed without merging**